### ISSUE TITLE
Ntriple reader gave error reading nt files including these characters.

### ISSUE THREAD

**Issue by afshinsadeghi** [SANSA-Stack/SANSA-RDF#24]
The triple reader gives error when the .nt file include one of these characters: \" or \\ or # or '

**Comment by GezimSejdiu** [comment]
Hi @afshinsadeghi ,
many thanks for opening this issue. Could you please be more specific (i.e which of the reader is throwing this error: this one [NTripleReader.load](https://github.com/SANSA-Stack/SANSA-RDF/blob/develop/sansa-rdf-spark-parent/sansa-rdf-spark-core/src/main/scala/net/sansa_stack/rdf/spark/io/NTripleReader.scala#L36 ) right?).

**Comment by afshinsadeghi** [comment]
Hi @GezimSejdiu yes, the  NTripleReader.scala#L25

**Comment by afshinsadeghi** [comment]
I have called it here: https://github.com/afshinsadeghi/SANSA-KGML/blob/8e47d50b2e2268f556527617f6ec746d76d7b83b/src/main/scala/net/sansa_stack/kgml/rdf/Main.scala#L95

**Comment by afshinsadeghi** [comment]
I tested reading these files:
mappingbased_objects_en.ttl.bz2   which is in
 http://downloads.dbpedia.org/current/core-i18n/en/mappingbased_objects_en.ttl.bz2

and
mappingbased_literals_en.ttl.bz2 which is in
 http://downloads.dbpedia.org/current/core-i18n/en/mappingbased_literals_en.ttl.bz2

**Comment by afshinsadeghi** [comment]
I have not the unprocessed nt files anymore, but I will reproduce them and upload them.  I will write here again when it is done.

**Comment by LorenzBuehmann** [comment]
@GezimSejdiu The file extension `ttl` as a bit misleading. The file format is in fact N-Triples, don't know why the DBpedia guys used `ttl` instead of `nt` especially as some tools guess the format by analyzing the file extension.
@afshinsadeghi I cannot reproduce the problem. I loaded [mappingbased_objects_en.ttl.bz2](http://downloads.dbpedia.org/current/core-i18n/en/mappingbased_objects_en.ttl.bz2) successfully (the extracted version indeed). Can you share the whole error stacktrace please? Actually, the parsing code should be pretty robust for N-Triples format as we're just splitting the input file(s) linewise and reuse the RIOT parser of the Apache Jena project. In addition, comment lines (starting with `#`) as well as empty lines will indeed be omitted.

**Comment by afshinsadeghi** [comment]
@LorenzBuehmann  Two more dataset that I tested were
http://resources.mpi-inf.mpg.de/yago-naga/yago3.1/yagoFacts.ttl.7z
http://resources.mpi-inf.mpg.de/yago-naga/yago3.1/yagoLabels.ttl.7z
These are real ttl that I need to upload nt versions.  They had invalid uni-code characters(not including number sign) in the list I wrote above. I am not sure, what will be the best policy for a reader, to say that the file has an error or to ignore those characters.

**Comment by LorenzBuehmann** [comment]
I can confirm that the YAGO dataset has some issues:

``` bash
mkdir /tmp/sansa-test
cd /tmp/sansa-test
wget http://resources.mpi-inf.mpg.de/yago-naga/yago3.1/yagoFacts.ttl.7z
7za x yagoFacts.ttl.7z
$JENA_HOME/bin/riot --time --check --out=NTRIPLES yagoFacts.ttl > yagoFacts.nt
```

a parse error occurs with
```
08:04:29 ERROR riot                 :: [line: 9597504, col: 8 ] Illegal unicode escape sequence value: \\ (0x5C)
```

> I am not sure, what will be the best policy for a reader, to say that the file has an error or to ignore those characters.
Ignoring a character is for sure no option, as this would invalidate the location target if the URI/IRI. The most fine-granular entity in RDF is an RDF resource, but the smallest statement is an RDF triple. Thus, the only option would be to ignore the whole RDF triple. Right now this isn't possible in SANSA.
I opened a ticket #27

**Comment by afshinsadeghi** [comment]
Seems to be a known issue
https://stackoverflow.com/questions/39664819/sanitize-yago-files-before-loading-into-apache-jena-tdb-triplestore/43515692 

As it is a cleaning issue of files and not reading them I assume my issue announced here closed and will follow up the #27

**Comment by LorenzBuehmann** [comment]
Summary:
- the first issue has been solved with release `0.3.0`
- the second issue is a general problem with poor data quality and needs some more rework and extension of the parser, refer to #27
